### PR TITLE
Properly handle non-generic string mapping types in unions and intersections

### LINKED ISSUE #57192: Inconsistent intersections of string literal types and pattern string mapping types
### 🔎 Search Terms

Literal type, string mapping type, intersection

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

https://www.typescriptlang.org/play?#code/C4TwDgpgBAYg9nKBeKBvKYBOcwC4oDOwmAlgHYDmUAvgNwBQ9okUAKgAzKwIDaARFhx8AurShQA9BKgB5ANZNw0VgEYu8OP0Fg+UAGRQABmAAkqIqUrUwh0eKmyFzZQCZ1vAdh36jp1ABk4AHcITABjAEMCCAAeC3IKAD5rWzFJaXlFFlYAZndNTyEfQJDwqNj4ykS7dKgAUUxsTAAaKCCI4AB+eiA

### 💻 Code

```ts
type Foo = { prop: string };

type T0 = Foo["prop"];  // string
type T1 = Foo["prop" & `p${string}p`];  // string
type T2 = Foo["prop" & `p${Lowercase<string>}p`];  // string
type T3 = Foo["prop" & Lowercase<string>];  // Error, wat?
```

### 🙁 Actual behavior

Error.

### 🙂 Expected behavior

No error.

### Additional information about the issue

See [here](https://github.com/microsoft/TypeScript/issues/57117#issuecomment-1912405637).

### PR DESCRIPTION
**Pull request by ahejlsberg**
Fixes #57192.

### CODE REVIEW

**Comment by ahejlsberg** [comment]
@typescript-bot test top100
@typescript-bot user test this
@typescript-bot run dt
@typescript-bot perf test this faster

**Comment by ahejlsberg** [comment]
@typescript-bot test top200

File: src/compiler/types.ts [hunk 1]
```diff
@@ -6125,6 +6125,8 @@ export const enum TypeFlags {
     NonPrimitive    = 1 << 26,  // intrinsic object type
     TemplateLiteral = 1 << 27,  // Template literal type
     StringMapping   = 1 << 28,  // Uppercase/Lowercase type
+    /** @internal */
+    Reserved1       = 1 << 29,  // Used by union/intersection type construction
```
Comment (DanielRosenwasser) [inline src/compiler/types.ts:6129]: This is kind of funky - I'd just directly make this `IncludesConstrainedTypeVariable`.

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
@typescript-bot test this

**Pull request merged**